src/App.tsx

```diff
@@ -1,17 +1,151 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { addTodo, deleteTodo, getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+
+import { TodoList } from './components/TodoList/TodoList';
+import { TempTodo } from './components/TempTodo/TempTodo';
+import { Footer } from './components/Footer/Footer';
+import { ErrorNotify } from './components/ErrorNotification/ErrorNotification';
+
+import { ErrorMessage } from './types/ErrorMessage';
+import { ErrorData } from './types/ErrorData';
+import { Sorting } from './types/Sorting';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isError, setIsError] = useState<ErrorData>({
+    status: false,
+    notification: ErrorMessage.None,
+  });
+  const [sortBy, setSortBy] = useState<Sorting>(Sorting.All);
+  const [title, setTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [deletedTodosIds, setDeletedTodosId] = useState<number[]>([]);
+
+  const visibleTodos = useMemo(() => {
+    switch (sortBy) {
+      case Sorting.Active:
+        return todos.filter(todo => !todo.completed);
+
+      case Sorting.Completed:
+        return todos.filter(todo => todo.completed);
+
+      default:
+        return todos;
+    }
+  }, [sortBy, todos]);
+
+  const completedTodosId = useMemo((
+    () => todos.filter(todo => todo.completed)
+      .map(todo => todo.id)
+  ), [todos]);
+
+  const handleError = useCallback((notification: ErrorMessage) => {
+    setIsError({
+      status: true,
+      notification,
+    });
+
+    setTimeout(() => setIsError({
+      status: false,
+      notification: ErrorMessage.None,
+    }), 3000);
+  }, []);
+
+  const loadTodos = async () => {
+    if (user) {
+      try {
+        const todosFromServer = await getTodos(user.id);
+
+        setTodos(() => todosFromServer);
```

Comment:
I think it also should work
```suggestion
        setTodos( todosFromServer);
```

Reply:
corrected